### PR TITLE
v1.0 with mapping

### PR DESCRIPTION
**Pull request by kaijli**
Successful mapping and v1.0 run at jaws id 79907
This is crucial for moving metaT forward.

### CODE REVIEW

File: annotation_full.wdl [hunk 1]
```diff
@@ -1,25 +1,26 @@
...
 
 workflow annotation {
+input {
   String  proj
   String  input_file
   String  imgap_project_id
   String  database_location="/refdata/img/"
   String  imgap_project_type="metagenome"
-  String?  gm_license="/refdata/licenses/.gmhmmp2_key"
+  String  gm_license="/refdata/licenses/.gmhmmp2_key"
```
Comment (aclum) [inline annotation_full.wdl]: This file should be left as optional, a user can also have the key in their homedir.

File: annotation_full.wdl [hunk 2]
```diff
@@ -263,38 +265,37 @@ task stage {
...
+  find_prefix=`grep ~{proj_id} ~{input_file} | head -1`
 
   set -euo pipefail
   if [[ $find_prefix ]]
   then
     echo "false" > run_map.txt
-    ln ${input_file} ${output_file} || ln -s ${input_file} ${output_file}
+    ln ~{input_file} ~{prefix}_map.fasta || ln -s ~{input_file} ~{prefix}_map.fasta
   else
     echo "true" > run_map.txt
```
Comment (aclum) [inline annotation_full.wdl]: we should always be generating the mapping file so I don't think we need this extra logic about find_prefix

File: functional-annotation.wdl [hunk 3]
```diff
@@ -1,50 +1,51 @@
...
-  String  last_container="microbiomedata/img-omics@sha256:d5f4306bf36a97d55a3710280b940b89d7d4aca76a343e75b0e250734bc82b71"
-
-  if(ko_ec_execute) {
+    input {
+        String  imgap_project_id
+        String  imgap_project_type
+        Int     additional_threads
+       # File    input_contigs_fasta
+        File    input_fasta
+        String  database_location
```
Comment (aclum) [inline functional-annotation.wdl]: default database location here should be /refdata/img/ to be consistent with other WDLs

File: prodigal.wdl
Comment (aclum) [inline prodigal.wdl:1]: This WDL can be deleted, it is obsolete

File: rfam.wdl [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
-  String claninfo_tsv="${database_location}"+"Rfam/13.0/Rfam.claninfo"
-  String feature_lookup_tsv="${database_location}"+"Rfam/13.0/Rfam_feature_lookup.tsv"
-  String container
-  
+    input {
+        String cmzscore
+        File imgap_input_fasta
+        String imgap_project_id
+        Int    additional_threads
+        String database_location="/refdata/img/"
```
Comment (aclum) [inline rfam.wdl:8]: this should be /refdata/ not /refdata/img/

File: structural-annotation.wdl [hunk 5]
```diff
@@ -1,72 +1,69 @@
...
+      Boolean prodigal_execute=true
+      Boolean genemark_execute=true
+      Boolean gff_and_fasta_stats_execute=true
+      String  database_location
+      String  container
+      String gm_license="/refdata/licenses/.gmhmmp2_key"
     }
-  }
 
-  if(trnascan_se_execute) {
```
Comment (aclum) [inline structural-annotation.wdl:38]: be consistent, if you are removing the boolean check also remove the boolean variable

File: test-small.wdl [hunk 6]
```diff
@@ -54,20 +58,22 @@ task prepare {
...
    File   struct_gff
    String url
    String proj
    String prefix = sub(proj, ":", "_")
+  }
 
-   command{
+   command <<<
        set -e
-       wget ${url}/${prefix}_functional_annotation.gff
```
Comment (aclum) [inline test-small.wdl:66]: update to pipefail used in other command blocks

File: annotation_full.wdl (hunk 1 above)
Comment (kaijli) [reply annotation_full.wdl]: If they have a key in their homedir, they can redefine the variable through json inputs. What are the advantages of having it as an optional variable vs pre-defined?

**Comment by kaijli** [comment]
New successful run at jaws id 79943 @aclum

**Review by aclum — APPROVED** [review]

**Pull request merged**